velox/common/base/VeloxException.cpp

```diff
@@ -71,29 +73,45 @@ VeloxException::VeloxException(
 
 namespace {
 
-// returns whether VeloxException stacktraces are enabled and whether, if they
-// are rate-limited, whether the rate-limit check passes
-bool isStackTraceEnabled() {
+/// returns whether VeloxException stacktraces are enabled and whether, if they
+/// are rate-limited, whether the rate-limit check passes
+bool isStackTraceEnabled(VeloxException::Type type) {
   using namespace std::literals::chrono_literals;
-
-  if (!FLAGS_velox_exception_stacktrace) {
-    // VeloxException stacktraces are disabled
+  const bool isSysException = type == VeloxException::Type::kSystem;
+  // TODO: deprecate 'FLAGS_deprecate velox_exception_stacktrace' flag once
+  // after 'FLAGS_velox_exception_user_stacktrace_enabled' and
+  // 'FLAGS_velox_exception_system_stacktrace_enabled' have been rolled out in
+  // production.
+  if (!FLAGS_velox_exception_stacktrace &&
+      ((isSysException && !FLAGS_velox_exception_system_stacktrace_enabled) ||
+       (!isSysException && !FLAGS_velox_exception_user_stacktrace_enabled))) {
+    // VeloxException stacktraces are disabled.
     return false;
   }
 
+  // TODO: deprecate 'FLAGS_velox_exception_stacktrace_rate_limit_ms' flag once
+  // after 'FLAGS_velox_exception_user_stacktrace_rate_limit_ms' and
+  // 'FLAGS_velox_exception_system_stacktrace_rate_limit_ms' have been rolled
+  // out in production.
+
+  const int32_t exceptionStackTraceRateLimitMs =
```

Comment:
nit: this variable is very long making the code hard to read; consider shortening to rateLimitMs